llvm/lib/Target/AMDGPU/AMDGPUMIRFormatter.cpp

```diff
@@ -17,6 +17,157 @@
 
 using namespace llvm;
 
+void AMDGPUMIRFormatter::printImm(raw_ostream &OS, const MachineInstr &MI,
+                      std::optional<unsigned int> OpIdx, int64_t Imm) const {
+
+  switch(MI.getOpcode()) {
+  case AMDGPU::S_DELAY_ALU:
+    assert(OpIdx == 0);
+    printSDelayAluImm(Imm, OS);
+    break;
+  default:
+    MIRFormatter::printImm(OS, MI, OpIdx, Imm);
+    break;
+  }
+}
+
+/// Implement target specific parsing of immediate mnemonics. The mnemonic is
+/// dot seperated strings.
```

Comment:
"Separated" here and elsewhere.

Also not sure what you mean by dot-separated. It looks more like an identifier with a leading dot.

Reply:
It's the same as the existing comment in the base class, although somehow I managed to introduce a mispelling in the process of copying it.

I agree it's not a great comment. I'll fix this version to more clear.